bundles/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveSwitchAllCommandClass.java

```diff
@@ -0,0 +1,244 @@
+	            logger.debug("NODE {}: Switch All report, device included in All On group.", (Object)this.getNode().getNodeId());
+	            break;
+	        case SWITCH_ALL_INCLUDE_OFF_ONLY:
+	            logger.debug("NODE {}: Switch All report, device included in All Off group.", (Object)this.getNode().getNodeId());
+	            break;
+	        case SWITCH_ALL_INCLUDE_ON_OFF:
+	            logger.debug("NODE {}: Switch All report, device included in All On and All Off group.", (Object)this.getNode().getNodeId());
+	            break;
+	        default:
+	            logger.warn(String.format("Unsupported Switch All mode 0x%02X.", mode));
+        }
+	        
+        SerialMessage result = new SerialMessage(this.getNode().getNodeId(), SerialMessage.SerialMessageClass.SendData, SerialMessage.SerialMessageType.Request, SerialMessage.SerialMessageClass.SendData, SerialMessage.SerialMessagePriority.Set);
+        byte[] newPayload = {
+        		(byte)this.getNode().getNodeId(), 
+        		3, 
+        		(byte) this.getCommandClass().getKey(), 
+        		(byte) SWITCH_ALL_SET, 
+        		(byte) mode };
+ 
+        result.setMessagePayload(newPayload);
+        return result;
+    }
+
+    /**
+     * Create the All On message
+     * @return
+     */
+    public SerialMessage allOnMessage() {
+        logger.debug("NODE {}: Switch All - Creating All On message.", (Object)this.getNode().getNodeId());
+        SerialMessage result = new SerialMessage(this.getNode().getNodeId(), SerialMessage.SerialMessageClass.SendData, SerialMessage.SerialMessageType.Request, SerialMessage.SerialMessageClass.SendData, SerialMessage.SerialMessagePriority.Set);
+        byte[] newPayload = { (byte)this.getNode().getNodeId(), 
+        		2, 
+        		(byte) this.getCommandClass().getKey(), 
+        		(byte) SWITCH_ALL_ON };
+        result.setMessagePayload(newPayload);
+        return result;
+    }
+
+    /**
+     * Create the All Off message
+     * @return
+     */
+    public SerialMessage allOffMessage() {
+        logger.debug("NODE {}: Switch All - Creating All Off message.", (Object)this.getNode().getNodeId());
+        SerialMessage result = new SerialMessage(this.getNode().getNodeId(), SerialMessage.SerialMessageClass.SendData, SerialMessage.SerialMessageType.Request, SerialMessage.SerialMessageClass.SendData, SerialMessage.SerialMessagePriority.Set);
+        byte[] newPayload = { (byte)this.getNode().getNodeId(), 
+        		2, 
+        		(byte) this.getCommandClass().getKey(), 
+        		(byte) SWITCH_ALL_OFF };
+        result.setMessagePayload(newPayload);
+        return result;
+    }
+
+    @Override
+    public boolean setOptions(ZWaveDbCommandClass options) {
+        if (options.isGetSupported != null) {
+            this.isGetSupported = options.isGetSupported;
+        }
+        return true;
+    }
+    
+    /**
+     * get the Switch All mode
+     * @return the mode
+     */
+    public int getMode() {
```

Comment:
I wonder if it wouldn't make sense to use an enum for `mode` to make it clearer for the upper layers.